snuba/consumers/consumer_builder.py

```diff
@@ -77,6 +91,7 @@ def __init__(
         max_batch_time_ms: int,
         metrics: MetricsBackend,
         parallel_collect: bool,
+        slice_id: Optional[int] = None,
```

Comment:
Let's remove the default here too, we want the real value

Reply:
I'm not sure if we would want to remove the default. A handful of different sliced functions (like `get_cluster`) rely on this behavior (only pass in a slice id if the storage set is sliced). This is also analogous to the way our settings work: only define in `SLICED_` settings if this is a sliced cluster/storage set/etc. If something is unsliced, define it in the regular settings.

Reply:
The consumer builder is only called in one place and we always pass a value from there. So why introduce the chance of calling it wrong / applying an incorrect default?

Reply:
ConsumerBuilder calls `get_cluster()` and the `slice_id` that is passed into the ConsumerBuilder in order to initialize it is the same as that which goes into the `get_cluster()` call. Are you suggesting that we change the behavior of `get_cluster` right now?

Reply:
I'm saying there was a reason why `get_cluster()` had to be that way with the default (which is that it's used in many non slice aware places right now). But I don't think that reason exists here.

Reply:
Oh, ok I think I'm starting to follow. Are you suggesting we keep it so that we explicitly pass in `None` for slice_id into ConsumerBuilder when necessary? And therefore, also into the Consumer CLI (via `--slice-id=None`)?

Reply:
The consumer CLI doesn't need to change anymore. Not passing any `--slice-id` ***is*** passing none. I think slice_id just should be passed in most of the places in this PR. There will be no real impact since we are already always passing the slice ID into most of the places anyway. But I think it helps prevent against incorrect usage in the future.

Reply:
Right, ok this makes sense.